src/runtime/initialize-component.ts

```diff
@@ -18,14 +18,11 @@ export const initializeComponent = async (
   Cstr?: any
 ) => {
   // initializeComponent
-  if (
-    (BUILD.lazyLoad || BUILD.hydrateServerSide || BUILD.style) &&
-    (hostRef.$flags$ & HOST_FLAGS.hasInitializedComponent) === 0
-  ) {
-    if (BUILD.lazyLoad || BUILD.hydrateClientSide) {
-      // we haven't initialized this element yet
-      hostRef.$flags$ |= HOST_FLAGS.hasInitializedComponent;
+  if ((hostRef.$flags$ & HOST_FLAGS.hasInitializedComponent) === 0) {
+    // Let the runtime know that the component has been initialized
+    hostRef.$flags$ |= HOST_FLAGS.hasInitializedComponent;
 
+    if (BUILD.lazyLoad || BUILD.hydrateClientSide) {
```

Comment:
So there are two `if` checks there actually. The top-most check _originally_ had the check containing `BUILD.hydrateServerSide`, but I removed the first part of that check because: 
1. The condition of `BUILD.style` was preventing that block of code to run for `dist-custom-elements` builds when the component did not have an associated stylesheet. But, if we just removed that portion of the check, then it would _never_ execute for `dist-custom-elements` builds
2. All components need to go through initialization anyway (AFAIK), so we should only really prevent that logic from executing if it has already been initialized

The second `if` statement is nested directly in that first block. I did not change that conditional or it's code block at all except for pulling out the line that sets the `hasInitializedComponent` flag on the hostRef since that exact code happened in both the `if` and `else` blocks.

Let me know if that doesn't make sense or you think my reasoning is wrong @rwaskiewicz

Reply:
Yep! That's exactly it!